#
# old_revision [20e2add42a314b4914e798757eb2a69ac3b491d0]
#
# patch "libpurple/protocols/jabber/parser.c"
#  from [6cecb079972daf216504b3dce358ca781c92c3d9]
#    to [9a47d8c1d7619d2d51db0f85502ee45208bd9e57]
#
============================================================
--- libpurple/protocols/jabber/parser.c	6cecb079972daf216504b3dce358ca781c92c3d9
+++ libpurple/protocols/jabber/parser.c	9a47d8c1d7619d2d51db0f85502ee45208bd9e57
@@ -132,6 +132,20 @@ jabber_parser_element_text_libxml(void *
 	xmlnode_insert_data(js->current, (const char*) text, text_len);
 }
 
+static void
+jabber_parser_
+static void
+jabber_parser_structured_error_handler(void * userData, xmlErrorPtr error)
+{
+	JabberStream *js = user_data;
+
+	purple_debug_error("jabber", "XML parser error for JabberStream %p: "
+								 "Domain %i, code %i, level %i: %s\n",
+					   js,
+					   error->domain, error->code, error->level,
+					   (error->message ? error->message : "(null)"));
+}
+
 static xmlSAXHandler jabber_parser_libxml = {
 	.internalSubset         = NULL,
 	.isStandalone           = NULL,
@@ -164,7 +178,7 @@ static xmlSAXHandler jabber_parser_libxm
 	._private               = NULL,
 	.startElementNs         = jabber_parser_element_start_libxml,
 	.endElementNs           = jabber_parser_element_end_libxml,
-	.serror                 = NULL
+	.serror                 = jabber_parser_structured_error_handler
 };
 
 void
